core-bundle/src/Routing/AbstractPageRouteProvider.php

```diff
@@ -0,0 +1,198 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\Routing;
+
+use Contao\CoreBundle\Framework\ContaoFramework;
+use Contao\Model\Collection;
+use Contao\PageModel;
+use Doctrine\DBAL\Connection;
+use Symfony\Cmf\Component\Routing\Candidates\CandidatesInterface;
+use Symfony\Cmf\Component\Routing\RouteProviderInterface;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\Routing\Route;
+
+abstract class AbstractPageRouteProvider implements RouteProviderInterface
+{
+    /**
+     * @var ContaoFramework
+     */
+    protected $framework;
+
+    /**
+     * @var Connection
+     */
+    protected $connection;
+
+    /**
+     * @var CandidatesInterface
+     */
+    protected $candidates;
+
+    public function __construct(ContaoFramework $framework, Connection $connection, CandidatesInterface $candidates)
+    {
+        $this->framework = $framework;
+        $this->connection = $connection;
+        $this->candidates = $candidates;
+    }
+
+    /**
+     * @return array<PageModel>
+     */
+    protected function findCandidatePages(Request $request): array
+    {
+        $candidates = $this->candidates->getCandidates($request);
+
+        if (empty($candidates)) {
+            return [];
+        }
+
+        $ids = [];
+        $aliases = [];
+
+        foreach ($candidates as $candidate) {
+            if (is_numeric($candidate)) {
+                $ids[] = (int) $candidate;
+            } else {
+                $aliases[] = $this->connection->quote($candidate);
+            }
+        }
+
+        $conditions = [];
+
+        if (!empty($ids)) {
+            $conditions[] = 'tl_page.id IN ('.implode(',', $ids).')';
+        }
+
+        if (!empty($aliases)) {
+            $conditions[] = 'tl_page.alias IN ('.implode(',', $aliases).')';
+        }
+
+        /** @var PageModel $pageModel */
+        $pageModel = $this->framework->getAdapter(PageModel::class);
+        $pages = $pageModel->findBy([implode(' OR ', $conditions)], []);
+
+        if (!$pages instanceof Collection) {
+            return [];
+        }
+
+        /** @var array<PageModel> */
+        return $pages->getModels();
+    }
+
+    /**
+     * @return array<int>
+     */
+    protected function getPageIdsFromNames(array $names): array
+    {
+        $ids = [];
+
+        foreach ($names as $name) {
+            if (0 !== strncmp($name, 'tl_page.', 8)) {
+                continue;
+            }
+
+            [, $id] = explode('.', $name);
+
+            if (!is_numeric($id)) {
+                continue;
+            }
```

Comment:
```suggestion
            [$table, $id] = explode('.', $name);

            if ('tl_page' !== $table || !is_numeric($id)) {
                continue;
            }
```

This would save the `strncmp()` call.